src/NServiceBus.Core/Licensing/LicenseExpiredForm.cs

```diff
@@ -1,145 +0,0 @@
-            }
-
-            Visible = true;
-        }
-
-        void browseButton_Click(object sender, EventArgs e)
-        {
-            using (var openDialog = new OpenFileDialog())
-            {
-                openDialog.InitializeLifetimeService();
-                openDialog.Filter = "License files (*.xml)|*.xml|All files (*.*)|*.*";
-                openDialog.Title = "Select License file";
-
-                var dialogResult = StaThreadRunner.ShowDialogInSTA(openDialog.ShowDialog);
-                if (dialogResult == DialogResult.OK)
-                {
-                    var licenseText = NonBlockingReader.ReadAllTextWithoutLocking(openDialog.FileName);
-                    try
-                    {
-                        LicenseVerifier.Verify(licenseText);
-                        var license = LicenseDeserializer.Deserialize(licenseText);
-
-                        if (LicenseExpirationChecker.HasLicenseExpired(license))
-                        {
-                            var message = "The license you provided has expired, select another file.";
-                            Logger.Warn(message);
-                            MessageBox.Show(this, message, "License expired", MessageBoxButtons.OK, MessageBoxIcon.Exclamation);
-                            return;
-                        }
-                        ResultingLicenseText = licenseText;
-                        Close();
-                    }
-                    catch (Exception exception)
-                    {
-                        var message = $"An error occurred when parsing the license.\r\nMessage: {exception.Message}\r\nThe exception details have been appended to the log.";
-                        Logger.Warn("Error parsing license", exception);
-                        MessageBox.Show(this, message, "Error parsing license", MessageBoxButtons.OK, MessageBoxIcon.Error);
-                    }
-                }
-            }
-        }
-
-        void PurchaseButton_Click(object sender, EventArgs e)
-        {
-            Process.Start("https://particular.net/licensing");
-        }
-
-        void getTrialLicenseButton_Click(object sender, EventArgs e)
-        {
-            string baseUrl;
-            if (CurrentLicense != null && !CurrentLicense.IsExtendedTrial)
-            {
-                // Original 14 day trial expired, give the user a chance to extend trial
-                baseUrl = "https://particular.net/extend-nservicebus-trial";
-            }
-            else
-            {
-                // Extended trial license expired, ask the user to Contact Sales
-                baseUrl = "https://particular.net/extend-your-trial-45";
-            }
-
-            var trialStart = TrialStartDateStore.GetTrialStartDate().ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
-            var url = $"{baseUrl}?NugetUser={IsNugetUser()}&PlatformInstaller={HasUserInstalledPlatform()}&TrialStartDate={trialStart}";
```

Comment:
And IsNugetUser is driven by information put in registry by [init.ps1](https://github.com/Particular/NServiceBus/blob/develop/packaging/nuget/tools/init.ps1). Does that die for .NET Core as well? Which means we'll also lose first-time install information?

Those are challenges we need to work through in smarter ways. Perhaps on the platform (and we may already be doing this) by setting a cookie and/or a Google Analytics event when the user downloads the platform.

But, those are reporting problems. The direct design of the page will depend much more on environment information like platform, version, etc, which we should definitely supply from `Environment.*` properties and not rely upon browser UserAgent, so that we don't need to take any dependencies on one of those databases like WURFL that are never up-to-date.